enterprise/server/hostedrunner/hostedrunner.go

```diff
@@ -195,6 +195,13 @@ func (r *runnerService) createAction(ctx context.Context, req *rnpb.RunRequest,
 		})
 	}
 
+	for k, v := range req.GetExecProperties() {
```

Comment:
since Go maps are non-deterministic, we might want to sort the platform properties by name after this for-loop, so that the snapshot cache key is deterministic (I think we blindly hash the Platform proto currently and the hash will be affected by the ordering of props)

(we could also directly use the Platform proto for the exec_properties field but I think that will probably make it less user-friendly, since Bazel uses a map for exec_properties)

Reply:
Done

What do you think about sorting and deduping the platform propreties in `snaploader` when we generate the key? It may invalidate some snapshots we have now, but will make the system more stable long term. (We could also have a short term migration period where we save snapshots under 2 keys - the existing key and a new one with the props sorted and deduped, and in a week we switch over to exclusively using the new key)